lnwallet/interface_test.go

```diff
@@ -1233,6 +1238,141 @@ func testSignOutputPrivateTweak(r *rpctest.Harness, w *lnwallet.LightningWallet,
 	}
 }
 
+func testReorgWalletBalance(r *rpctest.Harness, w *lnwallet.LightningWallet, t *testing.T) {
+	t.Logf("Running reorg wallet balance test")
+
+	// We first mine a few blocks to ensure any transactions still in the
+	// mempool confirm, and then get the original balance, before a
+	// reorganization that doesn't invalidate any existing transactions or
+	// create any new non-coinbase transactions. We'll then check if it's
+	// the same after the empty reorg.
+	_, err := r.Node.Generate(5)
+	if err != nil {
+		t.Fatalf("unable to generate blocks on passed node: %v", err)
+	}
+	// Give wallet time to catch up.
+	time.Sleep(time.Second)
```

Comment:
Rather than a hard coded sleep here (which may have weird issues when running on Travis), we can instead use the `IsSynced` method on the `WalletController` interface. The same swap can be repeated several times in this test case.